static/js/pathfinder.js

```diff
@@ -127,62 +123,59 @@ function reloadReports(){
                 $('#vulnerabilityReport').append('<option value="'+r.id+'">'+r.name+'</option>');
             }
         });
-    }
-    restRequest('POST', {'index':'reports'}, updateData, '/plugin/pathfinder/api');
-}
-
-function openSource(source_id){
-    viewSection('sources', '/advanced/sources');
-    setTimeout(function(s){ $('#profile-source-name').val(s).change(); }, 1000, source_id);
+    }).catch((error) => {
+        toast('Error reloading Pathfinder reports.', false);
+        console.error(error);
+    });
 }
 
 function checkScanStatus(){
-    function updateData(data){
-        number_finished = Object.keys(data.finished).length
-        number_failed = Object.keys(data.errors).length
-        if (data.pending.length == 0){
+    apiV2('POST', '/plugin/pathfinder/api', {'index':'status'}).then((response) => {
+        number_finished = Object.keys(response.finished).length
+        number_failed = Object.keys(response.errors).length
+        if (response.pending.length == 0){
```

Comment:
Another nitpick -- in general it's best to use `===` for equality comparisons instead of `==`. The triple equals compares both the value and the type of the variable, while the double equals will do a type conversion if necessary before comparing values (i.e. 2 == '2' -> `true`, 2 === '2' -> `false`). 

It's usually best to check if the variable you're comparing is also the type you're expecting, since JS isn't strongly typed and it can get really annoying to debug sometimes.

Reply:
Thank you got the tip, I've made this change throughout `pathfinder.js`.